pkg/app/daemon/daemon.go

```diff
@@ -94,19 +106,52 @@ func (self *rebaseDaemon) Run() error {
 }
 
 func (self *rebaseDaemon) writeTodoFile(path string) error {
-	todoContent := []byte(os.Getenv(RebaseTODOEnvKey))
+	if changeTodoActionEnvValue := os.Getenv(ChangeTodoActionEnvKey); changeTodoActionEnvValue != "" {
+		return self.changeTodoAction(path, changeTodoActionEnvValue)
+	} else {
+		todoContent := []byte(os.Getenv(RebaseTODOEnvKey))
+
+		prependLines := os.Getenv(PrependLinesEnvKey) != ""
+		if prependLines {
+			existingContent, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			todoContent = append(todoContent, existingContent...)
+		}
+
+		return os.WriteFile(path, todoContent, 0o644)
+	}
+}
 
-	prependLines := os.Getenv(PrependLinesEnvKey) != ""
-	if prependLines {
-		existingContent, err := os.ReadFile(path)
-		if err != nil {
+func (self *rebaseDaemon) changeTodoAction(path string, changeTodoActionEnvValue string) error {
+	lines := strings.Split(changeTodoActionEnvValue, "\n")
+	for _, line := range lines {
+		fields := strings.Split(line, ":")
+		if len(fields) != 2 {
+			return fmt.Errorf("Unexpected value for %s: %s", ChangeTodoActionEnvKey, changeTodoActionEnvValue)
+		}
+		sha, newAction := fields[0], self.actionFromString(fields[1])
+		if int(newAction) == 0 {
+			return fmt.Errorf("Unknown action in %s", changeTodoActionEnvValue)
+		}
+		if err := utils.EditRebaseTodo(path, sha, todo.Pick, newAction); err != nil {
 			return err
 		}
+	}
 
-		todoContent = append(todoContent, existingContent...)
+	return nil
+}
+
+func (self *rebaseDaemon) actionFromString(actionString string) todo.TodoCommand {
+	for t := todo.Pick; t < todo.Comment; t++ {
```

Comment:
Worth having a comment saying that we're iterating through all possible todo kinds (it's not obvious that Pick == 0 and Comment is the last one).

In fact, it would be good if the package itself could export a slice of all the different values so that we can iterate through it without needing to worry about it getting out of sync. But obvs not required for this PR

Reply:
This is obsolete, `actionFromString` is no longer needed after the json refactor.